streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -43,6 +47,9 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
   private val ssc = jobScheduler.ssc
   private val conf = ssc.conf
   private val graph = ssc.graph
+  private lazy val completedBothCheckpoints = new HashMap[Time, Boolean]()
+  private lazy val batchesGenerated = new TreeSet[Time](new TimeComparator)
+  private val noConcurrentJobs = ssc.sc.conf.getInt("spark.streaming.concurrentJobs", 1) == 1
```

Comment:
I dont like this, but it could be a necessary evil to make sure most users are unaffected by any change here.